src/trigger_in.h

```diff
@@ -3,17 +3,22 @@
 /** @file trigger_in.h
  * @brief Trigger in interface. */
 
-namespace ara { namespace com {
+#include "sm_types.h"
 
-/** @brief Fills [SWS_SM_91008] */
+namespace ara::com {
+
+/** @brief Fills [SWS_SM_91007] */
     class TriggerIn {
     public:
         bool IsTrigger() const;
+        sm::SMStateType GetDesiredState() const;
 
-        void SetTrigger(bool trigger);
+        void SetTrigger(sm::SMStateType);
+        void DiscardTrigger();
 
     private:
-        bool trigger;
+        bool isTrigger = false;
+        sm::SMStateType trigger;
```

Comment:
Please create another type for trigger containing sm::SMStateType and isTrigger. We should only have field 'trigger' according to requirements.

Reply:
Done